Sprint 15

socloudy-dev opened the pull request:
Задача 15 спринта завершена

socloudy-dev committed:
Добавлена работа с CoreData, исправлены недоработки в AddTrackerViewController с выбором цвета и эмодзи, поправлена работа фильтрации по дате, теперь на текщую дату можно отмечать трекер выполненным (ранее не работало при переключении на другую дату и обратно). Выполнена задача 15 спринта.

socloudy-dev committed:
Исправляю коммит, не были удалены файлы и не выгружен контроллер главного экрана

Kirchberg commented on Tracker/Trackers/Tracker-Main/TrackerCollectionViewCell.swift:
Конструкция избыточна, так как тернарный оператор со строковыми литералами никогда не вернет nil. Можно упростить до `let imageName = completed ? "Tracker Completed" : "Increase Counter"`

Kirchberg commented on Tracker/Trackers/AddTracker/AddTrackerViewController.swift:
```diff
@@ -12,9 +12,14 @@ final class AddTrackerViewController: UIViewController {
     private var tableTopToLabel: NSLayoutConstraint!
     
+    private let emojis = ["🙂","😻","🌺","🐶","❤️","😱","😇","😡","🥶","🤔","🙌","🍔","🥦","🏓","🥇","🎸","🏝","😪"]
+    private let colorsName = (1...18).map { String($0) }
```
Для названий цветов используется массив, сгенерированный как (1...18).map { String($0) }. Это работает, но опирается на "магические строки" названия ассетов должны строго совпадать с этими числами. Можно создать перечисление (enum) с этими цветами для лучшей читаемости

socloudy-dev committed:
З
аменил обращение к цветам трекера с массива чисел на enum, исправил избыточную конструкцию в TrackerCollectionViewCell.
